web-common/src/features/sources/modal/utils.ts

```diff
@@ -93,7 +93,14 @@ export function applyClickHouseCloudRequirements(
   connectorType: ClickHouseConnectorType,
   values: Record<string, unknown>,
 ): Record<string, unknown> {
-  if (connectorName === "clickhouse" && connectorType === "clickhouse-cloud") {
+  // Only force SSL for ClickHouse Cloud when the user is using individual params.
+  // DSN strings encapsulate their own protocol, so we should not inject `ssl` there.
+  const isDsnBased = "dsn" in values;
+  if (
+    connectorName === "clickhouse" &&
+    connectorType === "clickhouse-cloud" &&
+    !isDsnBased
+  ) {
```

Comment:
Nit: for complex conditionals like this, an explanatory variable would make the code easier-to-read. E.g.:

```typescript
const shouldEnforceSSL = connectorName === "clickhouse" && connectorType === "clickhouse-cloud" && !isDsnBased
if (shouldEnforceSSL) {...}
```